### PR TITLE
Custom losses

### PR DESCRIPTION
**Pull request by blengerich**
Separate out the prediction step from the loss step, and allow user to specify loss function when creating the regression model.

### CODE REVIEW

File: contextualized/regression.py [hunk 1]
```diff
@@ -416,6 +417,9 @@ def test_step(self, batch, batch_idx):
         self.log_dict({'test_loss': loss})
         return loss
 
+    def predict_from_sample_models(self, X, beta_hat, mu_hat):
```
Comment (cnellington) [inline contextualized/regression.py:420]: Make private and change name based on model-metamodel convention. `_predict_from_models` or `_model_predict` for example

**Review by cnellington — CHANGES_REQUESTED** [review]

File: contextualized/regression.py [hunk 2]
```diff
@@ -358,10 +358,11 @@ def forward(self, C, T):
 
 
 class ContextualizedRegressionBase(pl.LightningModule):
-    def __init__(self, *args, learning_rate=1e-3, link_fn=linear_fn(), **kwargs):
+    def __init__(self, learning_rate=1e-3, link_fn=linear_fn(), loss_fn=MSE, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (blengerich) [reply contextualized/regression.py:361]: Tests added in #60 and passing (`contextualized/regression/tests.py`). Requires passing dimensions by keyword rather than position, but I think that is okay. What do you think?

**Comment by blengerich** [comment]
Subsumed by #60
